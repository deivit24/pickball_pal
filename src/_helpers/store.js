import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
