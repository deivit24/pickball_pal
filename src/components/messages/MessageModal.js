import React, { useState } from 'react';

import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import MessageForm from './MessageForm';
import Modal from '@material-ui/core/Modal';
import Button from '@material-ui/core/Button';
import Backdrop from '@material-ui/core/Backdrop';
import { useSpring, animated } from 'react-spring/web.cjs'; // web.cjs is required for IE 11 support

const useStyles = makeStyles((theme) => ({
  modal: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  paper: {
    backgroundColor: theme.palette.background.paper,
    border: '1px solid #fff',
    boxShadow: theme.shadows[5],
    padding: theme.spacing(2, 4, 3),
  },
}));

const Fade = React.forwardRef(function Fade(props, ref) {
  const { in: open, children, onEnter, onExited, ...other } = props;
  const style = useSpring({
    from: { opacity: 0 },
    to: { opacity: open ? 1 : 0 },
    onStart: () => {
      if (open && onEnter) {
        onEnter();
      }
    },
    onRest: () => {
      if (!open && onExited) {
        onExited();
      }
    },
  });

  return (
    <animated.div ref={ref} style={style} {...other}>
      {children}
    </animated.div>
  );
});

Fade.propTypes = {
  children: PropTypes.element,
  in: PropTypes.bool.isRequired,
  onEnter: PropTypes.func,
  onExited: PropTypes.func,
};

const MessageModal = ({ messageMe, display, name, id, handleSubmit }) => {
  const classes = useStyles();
  const [open, setOpen] = useState(false);

  const handleOpen = (e) => {
    e.preventDefault();
    handleSubmit();
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <>
      <Button
        className={`mx-auto ${display}`}
        variant="contained"
        color="primary"
        disabled={messageMe}
        onClick={handleOpen}
      >
        Message Me
      </Button>
      <Modal
        aria-labelledby="spring-modal-title"
        aria-describedby="spring-modal-description"
        className={classes.modal}
        open={open}
        onClose={handleClose}
        closeAfterTransition
        BackdropComponent={Backdrop}
        BackdropProps={{
          timeout: 500,
        }}
      >
        <Fade in={open}>
          <div className={classes.paper}>
            <h2 id="spring-modal-title">{name}</h2>
            <MessageForm replyId={id} handleClose={handleClose} />
          </div>
        </Fade>
      </Modal>
    </>
  );
};

export default MessageModal;
