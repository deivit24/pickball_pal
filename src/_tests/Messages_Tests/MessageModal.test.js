import React from 'react';
import { render } from '@testing-library/react';
import MessageModal from '../../components/messages/MessageModal';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing MessageModal

describe('Testing the MessageModal', () => {
  const id = '0001';
  const messageMe = false;
  const display = '';
  const name = 'David Salazar';

  const handleSubmit = () => {
    console.log('Testing Handle Submit');
  };
  it('it renders MessageModal without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <MessageModal
            id={id}
            handleSubmit={handleSubmit}
            name={name}
            display={display}
            messageMe={messageMe}
          />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <MessageModal
            id={id}
            handleSubmit={handleSubmit}
            name={name}
            display={display}
            messageMe={messageMe}
          />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
