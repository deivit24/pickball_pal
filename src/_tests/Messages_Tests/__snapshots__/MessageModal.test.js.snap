// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the MessageForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="MessageForm"
  >
    <form
      action=""
      autocomplete="off"
      novalidate=""
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline mr-2 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-multiline MuiInput-multiline"
      >
        <textarea
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          id="outlined-multiline-flexible"
          name="message"
          placeholder="message"
          rows="4"
        />
      </div>
      <br />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained mt-3  MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Send
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Testing the MessageModal matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained mx-auto  MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Message Me
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
